pil_analyzer/src/pil_analyzer.rs

```diff
@@ -191,6 +191,38 @@ impl<T: FieldElement> PILContext<T> {
             PilStatement::ConstantDefinition(_, name, value) => {
                 self.handle_constant_definition(name, value)
             }
+            PilStatement::LetStatement(start, name, None) => {
+                // Handle all let statements without assignment as witness column declarations for now.
+                self.handle_polynomial_declarations(
+                    self.to_source_ref(start),
+                    vec![PolynomialName {
+                        name,
+                        array_size: None,
+                    }],
+                    PolynomialType::Committed,
+                )
+            }
+            PilStatement::LetStatement(start, name, Some(value)) => {
+                // Determine this is a fixed column or a constant depending on the structure
+                // of the value.
+                // Later, this should depend on the type.
```

Comment:
Do I understand this right, that in the future you could do `let x: Witness;` to be explicit about the type again?

Reply:
No, I would still leave some slack here. In the future you could do `let x: col` (`col` is synonym for `int -> fe`), but if you do
```
let x: col;
x = |i| i as fe;
```
then the compile might still think `x` to be defined well enough to be considered a fixed column. but I would not do that for not.